svg.go

```diff
@@ -0,0 +1,62 @@
+package jujusvg
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/juju/xml"
+)
+
+const svgNamespace = "http://www.w3.org/2000/svg"
+
+// Process an icon SVG file from a reader, removing anything surrounding
+// the <svg></svg> tags, which would be invalid in this context (such as
+// <?xml...?> decls, directives, etc), writing out to a writer.  In
+// addition, loosely check that the icon is a valid SVG file.
+func processIcon(r io.Reader, w io.Writer) error {
+	dec := xml.NewDecoder(r)
+	dec.DefaultSpace = svgNamespace
+
+	enc := xml.NewEncoder(w)
+
+	svgStartFound := false
```

Comment:
I think this will fail break if the the svg has <svg> elements not at the top level. I think you need to count the depth of svg elements just to be sure.
```go
depth := 0
for depth < 1 {
	tok, err := dec.Token()
	if err != nil {
		if err == io.EOF {
			break
		}
		return fmt.Errorf("cannot get token: %v", err)
	}
	tag, ok := tok.(xml.StartElement)
	if ok && tag.Name.Space == svgNamespace && tag.Name.Local == "svg" {
		depth++
		if err := enc.EncodeToken(tok); err != nil {
			return fmt.Errorf("cannot encode token %#v: %v", tok, err)
		}
	}
}
for depth > 0 {
	tok, err := dec.Token()
	if err != nil {
		if err == io.EOF {
			break
		}
		return fmt.Errorf("cannot get token: %v", err)
	}
	switch tag := tok.(type) {
	case xml.StartElement:
		if tag.Name.Space == svgNamespace && tag.Name.Local == "svg" {
			depth++
		}
	case xml.EndElement:
		if tag.Name.Space == svgNamespace && tag.Name.Local == "svg" {
			depth--
		}
	}
	if err := enc.EncodeToken(tok); err != nil {
		return fmt.Errorf("cannot encode token %#v: %v", tok, err)
	}
}
```